### PR TITLE
Add stand-alone player to playground

### PR DESCRIPTION
**Pull request by spectranaut**
### Proposed Changes

This PR adds a experimental "standalone player" to the GUI's playground. The `standalone-player.html` is example of how to create minimalist scratch project player without the editor view. I hoping to get feedback and continue to work to make the example more robust -- this is just a first draft! I'll also be looking to contribute back to the various repos to make including a player for scratch projects easier to embed in websites.

### Reason for Changes

There is no standard or recommended way to use the scratch player alone.

### Test Coverage

No tests, this is just a draft.

### CODE REVIEW

File: src/lib/storage.js [hunk 1]
```diff
@@ -10,6 +13,8 @@ class Storage extends ScratchStorage {
     constructor () {
         super();
         this.cacheDefaultProject();
+        this.projectHost = DEFAULT_PROJECT_SERVER;
+        this.assetHost = DEFAULT_ASSET_SERVER;
```
Comment (spectranaut) [inline src/lib/storage.js]: Is there a reason not to do this? In scratch-gui code, `setAssetHost` and `setProjectHost` will be called and overwrite this value (with the same value, in the scratch-gui case).

The standalone player could explicitly set these values as well, but having a default lowers the start up cost/boiler plate code for the standalone player scenario.

**Comment by towerofnix** [comment]
:tada:! This will definitely be a handy reference for the many developers who frequently ask how to embed Scratch in their program or website.

File: src/playground/standalone-player.jsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    vm.attachStorage(storage);
+
+    // Compatibility mode will set the frame rate to 30 TPS,
+    // which is the standard for the scratch player.
+    vm.setCompatibilityMode(true);
+
+    vm.downloadProjectId(EXAMPLE_PROJECT);
+
+    vm.on('workspaceUpdate', () => {
+        setTimeout(() => vm.greenFlag(), 1000);
```
Comment (mzgoddard) [inline src/playground/standalone-player.jsx]: I don't think this should autoplay. Some user input should be required.

File: src/playground/standalone-player.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    // Initialize storage
+    storage.addOfficialScratchWebStores();
+    vm.attachStorage(storage);
+
+    // Compatibility mode will set the frame rate to 30 TPS,
+    // which is the standard for the scratch player.
+    vm.setCompatibilityMode(true);
+
+    vm.downloadProjectId(EXAMPLE_PROJECT);
```
Comment (mzgoddard) [inline src/playground/standalone-player.jsx]: Why is this hard coded?

File: src/playground/standalone-player.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        // E.g., prevent scroll.
+        if (e.target !== document && e.target !== document.body) {
+            e.preventDefault();
+        }
+    });
+
+    // Run threads
+    vm.start();
+
+    vm.greenFlag();
```
Comment (mzgoddard) [inline src/playground/standalone-player.jsx]: Do we need to green flag here?

File: src/playground/standalone-player.jsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    vm.attachV2BitmapAdapter(new BitmapAdapter());
+    vm.attachV2SVGAdapter(new SVGRenderer());
+
+    // Initialize storage
+    storage.addOfficialScratchWebStores();
+    vm.attachStorage(storage);
+
+    // Compatibility mode will set the frame rate to 30 TPS,
+    // which is the standard for the scratch player.
+    vm.setCompatibilityMode(true);
```
Comment (spectranaut) [inline src/playground/standalone-player.jsx]: Why isn't this the default? Took me and @z a minute to figure out why everything was sped up (before adding this).

File: src/playground/standalone-player.jsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import VirtualMachine from 'scratch-vm';
+import RenderWebGL from 'scratch-render';
+import AudioEngine from 'scratch-audio';
+import {SVGRenderer, BitmapAdapter} from 'scratch-svg-renderer';
+import storage from '../lib/storage';
+
+// This file is an example of how to create a standalone, full screen
+// minimal scratch player without the editor view.
+
+const EXAMPLE_PROJECT = 10128067; // 'Make it Dance' Project
```
Comment (mzgoddard) [inline src/playground/standalone-player.jsx]: I think we want the example project to follow the default the other scratch-gui pages use. The Scratch Cat default project id 0.

File: src/lib/storage.js (hunk 1 above)
Comment (mzgoddard) [reply src/lib/storage.js]: This makes it easier. I think though its one of those slippery slopes kind of deals. We probably don't want a default configuration for scratch that is distributed over Scratch's implementation?

File: src/playground/standalone-player.jsx (hunk 4 above)
Comment (spectranaut) [reply src/playground/standalone-player.jsx]: nope

File: src/playground/standalone-player.jsx (hunk 2 above)
Comment (spectranaut) [reply src/playground/standalone-player.jsx]: I added the green flag overlay -- click to start :)

**Comment by benjiwheeler** [comment]
Thanks for this, @spectranaut ! We're also starting work on a related feature for the website's project pages, that will give you the code you need to embed a given project. 

See: https://github.com/LLK/scratch-www/issues/2905

File: src/playground/standalone-player.jsx (hunk 5 above)
Comment (spectranaut) [reply src/playground/standalone-player.jsx]: Made an issue to discuss this: https://github.com/LLK/scratch-vm/issues/2137

**Comment by spectranaut** [comment]
> Code LG. I think it would probably be good to isolate the storage config even if it means duplicating it, just to keep this a truly standalone demo.

Done!

**Comment by spectranaut** [comment]
closing this in favor of landing in the VM playground! Thanks for everyones help and support here :)

new PR: https://github.com/LLK/scratch-vm/pull/2156

**Comment by alexwenbj** [comment]
@spectranaut Hi,This is useful for me.Just one problem, it doesn't support the "`ask ... and wait`"  and "`monitor the variable`".Any idea for these?

**Pull request closed without merging**